tests/test_slurm_command_gen_strategy.py

```diff
@@ -185,6 +185,47 @@ def test_generate_full_srun_command(self, strategy_fixture: SlurmCommandGenStrat
         full_srun_command = strategy_fixture.generate_full_srun_command({}, {}, {}, "")
         assert full_srun_command == " \\\n".join(["srun", "--test", "test_arg", "test_command"])
 
+    def test_generate_full_srun_command_with_run_pre_test_flag(
+        self, jax_strategy_fixture: JaxToolboxSlurmCommandGenStrategy, slurm_system: SlurmSystem
+    ):
+        # Assign the slurm_system to the strategy fixture
+        jax_strategy_fixture.slurm_system = slurm_system
+
+        # Mock necessary methods and attributes
```

Comment:
Would it make sense moving this code into the fixture itself? This will make the test easier to read and work with. Assert could be transformed into `assert jax_strategy_fixture._generate_pre_test_command.return_value in result`.

Reply:
Can be done. Would move this discussion to seperate refactor PR as we discussed in the call. Unit tests unless broken or doesn't provide coverage shouldn't block the feature request. If there is anything functionally broken, let me know. Happy to add it.